fem/libceed/util.hpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_LIBCEED_UTIL
+#define MFEM_LIBCEED_UTIL
+
+#include "../../config/config.hpp"
+#ifdef MFEM_USE_CEED
+#include <ceed.h>
+#include <ceed-hash.h>
+#endif
+#include <tuple>
+#include <unordered_map>
+#include <string>
+
+namespace mfem
+{
+
+class FiniteElementSpace;
+class IntegrationRule;
+class Vector;
+
+/** @brief Function that determines if a CEED kernel should be used, based on
+    the current mfem::Device configuration. */
+bool DeviceCanUseCeed();
+
+namespace ceed
+{
+
+/** @brief Remove from ceed_basis_map and ceed_restr_map the entries associated
+    with the given @a fes. */
+void RemoveBasisAndRestriction(const mfem::FiniteElementSpace *fes);
+
+#ifdef MFEM_USE_CEED
+
+/// Initialize a CeedVector from a Vector
+void InitVector(const mfem::Vector &v, CeedVector &cv);
+
+/// Initialize a strided CeedElemRestriction
```

Comment:
This comment doesn't tell you any more than a name. Maybe describe the inputs, or at least refer to the libCEED documentation.

Reply:
I improved the documentation, tell me if you like it.